UNRELEASED.md

```diff
@@ -11,6 +11,7 @@ Use [the changelog guidelines](https://git.io/polaris-changelog-guidelines) to f
 ### Enhancements
 
 ### Bug fixes
+- Fixed animation for Modal when being rendered asynchronously ([#2076](https://github.com/Shopify/polaris-react/pull/2076))
```

Comment:
I would probably just say 'when being mounted with `open:true` directly' or something like that?
Rendering the modal asynchronously using `react-async` is one of the scenario where it's mounted directly with `open:true` but there are others.

Reply:
Great catch, thanks!